### PR TITLE
feat: Add support for PostgreSQL User Role

### PR DESCRIPTION
**Pull request by oxyno-zeta**
- Add support for PostgresqlUserRole
- Deprecate PostgresqlUser
- Fix Helm chart
- Correct context usage
- Upgrade linter & fix errors

### CODE REVIEW

File: config/crd/bases/postgresql.easymile.com_postgresqluserroles.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                description: User role prefix
+                type: string
+              userPasswordRotationDuration:
+                description: User password rotation duration
+                type: string
+              workGeneratedSecretName:
+                description: Simple user password tuple generated secret name
+                type: string
+            required:
+            - privileges
```
Comment (mmauro-ez) [inline config/crd/bases/postgresql.easymile.com_postgresqluserroles.yaml:110]: Mode is also required

File: apis/postgresql/v1alpha1/postgresqluserrole_types.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	// +kubebuilder:validation:Enum=PROVIDED;MANAGED
+	Mode ModeEnum `json:"mode,omitempty"`
+	// Privileges
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Min=1
+	Privileges []*PostgresqlUserRolePrivilege `json:"privileges"`
+	// User role prefix
+	// +optional
+	RolePrefix string `json:"rolePrefix,omitempty"`
```
Comment (mmauro-ez) [inline apis/postgresql/v1alpha1/postgresqluserrole_types.go:66]: This is ambiguous to have required fields only in one mode.

It seems the only solution is validation rules, but only available in k8s 1.25 (in beta...)
https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules

File: config/crd/bases/postgresql.easymile.com_postgresqluserroles.yaml (hunk 1 above)
Comment (oxyno-zeta) [reply config/crd/bases/postgresql.easymile.com_postgresqluserroles.yaml:110]: It is generated so... But Enums are required :)

File: apis/postgresql/v1alpha1/postgresqluserrole_types.go (hunk 2 above)
Comment (oxyno-zeta) [reply apis/postgresql/v1alpha1/postgresqluserrole_types.go:66]: No better solution for the moment...

File: controllers/postgresql/postgres/aws.go [hunk 3]
```diff
@@ -111,5 +111,55 @@ func (c *awspg) DropRole(role, newOwner, database string) error {
...
+	if err != nil {
+		// Try to cast error
+		pqErr, ok := err.(*pq.Error)
+		if !ok {
+			return err
+		}
+
+		if pqErr.Code == RoleNotFoundErrorCode {
+			// The group role does not exist, no point of granting roles
+			c.log.Info(fmt.Sprintf("not granting %s to %s as %s does not exist", role, newOwner, newOwner))
```
Comment (remifabas) [inline controllers/postgresql/postgres/aws.go:147]: Maybe change log level to Warn ? Or RoleNotFound is a regular case ?

Comment (oxyno-zeta) [reply controllers/postgresql/postgres/aws.go:147]: Yes maybe but as this code have been copy pasted upper too, I think this must be changed in another PR. What do you think ?

**Review by mmauro-ez — APPROVED** [review]

File: controllers/postgresql/postgres/aws.go (hunk 3 above)
Comment (remifabas) [reply controllers/postgresql/postgres/aws.go:147]: agreed :+1:

**Review by remifabas — APPROVED** [review]

**Pull request merged**